Cython/Compiler/Nodes.py

```diff
@@ -2637,8 +2637,11 @@ def generate_argument_parsing_code(self, env, code):
         def put_into_closure(entry):
             if entry.in_closure and not arg.default:
                 code.putln('%s = %s;' % (entry.cname, entry.original_cname))
-                code.put_var_incref(entry)
-                code.put_var_giveref(entry)
+                if entry.type.is_memoryviewslice:
+                    code.put_incref_memoryviewslice(entry.cname, True)
```

Comment:
This is probably what was meant here.
```suggestion
                    code.put_incref_memoryviewslice(entry.cname, have_gil=True)
```